synchronizer/state_tx_reuse.go

```diff
@@ -2,42 +2,55 @@ package synchronizer
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/0xPolygonHermez/zkevm-synchronizer-l1/log"
 )
 
+// object: ReuseStateTxProvider
+// intent:
+//   for a process that create a new tx each time
+//   you can bypass this behavious returning a 'fake tx' ReuseStateTx
+//   that ignore commit. The rollback is real
+//   The usage implies that the level that use this
+//   will commit / rollback the real Tx
+// It substitute the state class that implements
+//    BeginTransaction
+
```

Comment:
can you explain i which cases this is needed? Seems a bit dangerous

Reply:
This is used to sync  events previous to genesis (needs for other rollups that starts after LxLy deployment). It doesn't store l1block so it need to be atomic to be sure that have been executed totally